code/modules/mob/living/simple_animal/revenant/revenant.dm

```diff
@@ -215,17 +222,17 @@
 
 /datum/objective/revenant/New()
 	targetAmount = rand(100,200)
-	explanation_text = "Absorb [targetAmount] points of essence."
+	explanation_text = "Absorb [targetAmount] points of essence from humans."
 	..()
 
 /datum/objective/revenant/check_completion()
-	if(!istype(owner.current, /mob/living/simple_animal/revenant) || !owner.current)
+	if(!istype(owner, /mob/living/simple_animal/revenant) || !owner)
```

Comment:
No.
```owner``` is a ```/datum/mind```, never a revenant
the check for ```owner.current``` was correct.

also remove the ```||``` part of this check, everything after the ```||``` is redundant.